flight/Modules/Actuator/actuator.c

```diff
@@ -714,31 +612,180 @@ static bool set_channel(uint8_t mixer_channel, float value, const ActuatorSettin
+		PIOS_Servo_SetMode(actuatorSettings.TimerUpdateFreq, actuatorSettings.TimerPwmResolution, ACTUATORSETTINGS_TIMERPWMRESOLUTION_NUMELEM);
 	}
 }
 
-static void ActuatorSettingsUpdatedCb(UAVObjEvent * ev)
+static void SettingsUpdatedCb(UAVObjEvent * ev)
 {
-	actuator_settings_updated = true;
+	settings_updated = true;
 }
 
-static void MixerSettingsUpdatedCb(UAVObjEvent * ev)
+static float MixChannel(int ct, ActuatorDesiredData *desired,
+		float curve1, float curve2)
 {
-	mixer_settings_updated = true;
+	MixerSettingsMixer1TypeOptions type = GetMixerType(ct);
+
+	switch (type) {
+		case MIXERSETTINGS_MIXER1TYPE_DISABLED:
+			// Set to minimum if disabled.  This is not the same as saying PWM pulse = 0 us
+			return -1;
+			break;
+
+		case MIXERSETTINGS_MIXER1TYPE_SERVO:
+			return ProcessMixer(ct, curve1, curve2, desired);
+			break;
+
+		case MIXERSETTINGS_MIXER1TYPE_MOTOR:
+			(void) 0;	// nil statement
+			float val = ProcessMixer(ct, curve1, curve2, desired);
+
+			if (val > 0) {
+				// Apply curve fitting, mapping the input to the propeller output.
+				val = actuatorSettings.MotorInputOutputCurveFit[ACTUATORSETTINGS_MOTORINPUTOUTPUTCURVEFIT_A] *
+					powf(val, actuatorSettings.MotorInputOutputCurveFit[ACTUATORSETTINGS_MOTORINPUTOUTPUTCURVEFIT_B]);
+			} else {
+				// Idle throttle
+				val = 0.0f;
+			}
+
+			return val;
+			break;
+		// If an accessory channel is selected for direct bypass mode
+		// In this configuration the accessory channel is scaled and
+		// mapped directly to output.
+		// Note: THERE IS NO SAFETY CHECK HERE FOR ARMING
+		// these also will not be updated in failsafe mode.  I'm not sure
+		// what the correct behavior is since it seems domain specific.
+		// I don't love this code
+		case MIXERSETTINGS_MIXER1TYPE_ACCESSORY0:
+		case MIXERSETTINGS_MIXER1TYPE_ACCESSORY1:
+		case MIXERSETTINGS_MIXER1TYPE_ACCESSORY2:
+		case MIXERSETTINGS_MIXER1TYPE_ACCESSORY3:
+		case MIXERSETTINGS_MIXER1TYPE_ACCESSORY4:
+		case MIXERSETTINGS_MIXER1TYPE_ACCESSORY5:
+			(void) 0;
+
+			AccessoryDesiredData accessory;
+			if(AccessoryDesiredInstGet(type - MIXERSETTINGS_MIXER1TYPE_ACCESSORY0,&accessory) == 0)
+				return accessory.AccessoryVal;
+			else
+				return -1;
+			break;
+		case MIXERSETTINGS_MIXER1TYPE_CAMERAPITCH:
+		case MIXERSETTINGS_MIXER1TYPE_CAMERAROLL:
+		case MIXERSETTINGS_MIXER1TYPE_CAMERAYAW:
+			(void) 0;
+
+			CameraDesiredData cameraDesired;
+			if( CameraDesiredGet(&cameraDesired) == 0 ) {
+				switch(type) {
+					case MIXERSETTINGS_MIXER1TYPE_CAMERAROLL:
+						return cameraDesired.Roll;
+					case MIXERSETTINGS_MIXER1TYPE_CAMERAPITCH:
+						return cameraDesired.Pitch;
+					case MIXERSETTINGS_MIXER1TYPE_CAMERAYAW:
+						return cameraDesired.Yaw;
+					default:
+						break;
+				}
+			}
+			return -1;
+		/* No default-- make sure we handle all types positively here */
+	} 
+
+	/* Can't get here. */
+	return -1;
+}
+
+static MixerSettingsMixer1TypeOptions GetMixerType(int idx) 
```

Comment:
I was all excited when I saw you removed my code above because I thought you had come up with some awesome way to get rid of the ugliness I introduced. So sad to discover it's not gone yet! :D